src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,231 @@
+
+        When job is queued, squeue command will return code 0 and out put will be like (with no job info just the header):
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+
+        When job is running, squeue command will return code 0 and out put will be like:
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+        5987779 regular_m submit_t      tgu  R       1:20      2 nid[005689-005690]
+
+        When job is finished/failed:
+            * job status is cached for a period of time (about 10 mins). squeue command will return code 0 and
+            out put will be like below (with no job info just the header):
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+            * after the period of time, squeue command will return error code 1 and error message:
+            slurm_load_jobs error: Invalid job id specified
+
+        Matrix of job status:
+        Done file  Exit Code   Job listed    Status
+        Y          Any         Any           COMPLETED
+        N          0           N             PENDING or FAILED (if failed, exit code will flip to -1 within ~10m)
+        N          0           Y             RUNNING
+        N          -1          N/A           FAILED
+
+        NOTE: If job fails and squeue command hasn't flipped the exit code to -1, this function will return 'PENDING'
+        as the job status. After about 10 mins, the exit code will be flipped to -1 and this function will return
+        'FAILED' status as the job status.
+        """
+
+        # Once all tasks are completed, NERSC generates this file.
+        # (ref: https://docs.nersc.gov/jobs/workflow/taskfarmer/#output)
+        done_file = os.path.join(self.job_dir, NERSC_SLURM_DONE_FILE)
+        if os.path.isfile(done_file):
+            return JobStatus.COMPLETED
+
+        # check job status using squeue command
+        std_out_file, std_err_file, exit_code = tf_common.run_nersc_command(
+            ['squeue', '-j', str(job_id)], self.job_dir, log_file_prefix='squeue', check_return_code=False)
+
+        if exit_code != 0:
+            # job finished without creating the done file indicating the job is failed
+            return JobStatus.FAILED
+
+        with open(std_out_file, "r") as std_out, open(std_err_file, "r") as std_err:
+            squeue_out = std_out.read().strip()
+            if str(job_id) in squeue_out:
+                return JobStatus.RUNNING
+            else:
+                return JobStatus.PENDING
+
+    def _append_task_info(self, task_info):
+        """
+        Append the task information to the task info file
+        """
+
+        task_info.update({'kbase_collection': self.kbase_collection, 'load_ver': self.load_ver, 'tool': self.tool})
+
+        if not all(key in task_info for key in REQUIRED_TASK_INFO_KEYS):
+            raise ValueError(f"task_info must contain all keys: {REQUIRED_TASK_INFO_KEYS}")
```

Comment:
I would remove the 3 guaranteed keys from REQUIRED_TASK... otherwise this error message is going to be confusing

Reply:
👍

Reply:
:+1: